airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -599,6 +586,16 @@ public void writeStandardSync(final StandardSync standardSync) throws JsonValida
     persistence.writeConfig(ConfigSchema.STANDARD_SYNC, standardSync.getConnectionId().toString(), standardSync);
   }
 
+  public ScheduleType getScheduleType(final UUID connId) throws IOException {
+    return database.query(ctx -> ctx.select(CONNECTION.SCHEDULE_TYPE).from(CONNECTION)
```

Comment:
I don't think I understand how `ctx` works here.
I get that it's a context
It seems like we are doing a SQL-like query on the connection table for the column `schedule_type` as defined in `CONNECTION.SCHEDULE_TYPE` from the table `CONNECTION`. 

please tell me if those assumptions are invalid